larray/viewer/view.py

```diff
@@ -1155,9 +1155,12 @@ def setup_and_check(self, data, title='', readonly=False, minvalue=None, maxvalu
         Setup MappingEditor:
         return False if data is not supported, True otherwise
         """
-        if not isinstance(data, la.Session):
-            data = la.Session(data)
-        self.data = data
+        if data is None or isinstance(data, str):
```

Comment:
I would rather move this block of code (and probably the `self.add_list_items(arrays)` line) to the end of the method so that you can handle the `isinstance(data, str)` case directly. Otherwise, the `self.data = la.Session()` **looks** like it is wrong. I do not think it is, but I certainly thought it was when I first saw that line.